qiskit_ibm_runtime/utils/qctrl.py

```diff
@@ -0,0 +1,143 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Qctrl validation functions and helpers."""
+
+import logging
+from typing import Any, Optional, Dict, List
+
+from ..options import Options
+from ..options import EnvironmentOptions, ExecutionOptions, TranspilationOptions, SimulatorOptions
+
+logger = logging.getLogger(__name__)
+
+
+def validate(options: Dict[str, Any]) -> None:
+    """Validates the options for qctrl"""
+
+    # Raise error on bad options.
+    _raise_if_error_in_options(options)
+    # Override options and warn.
+    _warn_and_clean_options(options)
+
+    # Default validation otherwise.
+    TranspilationOptions.validate_transpilation_options(options.get("transpilation"))
+    execution_time = options.get("max_execution_time")
+    if not execution_time is None:
+        if (
+            execution_time < Options._MIN_EXECUTION_TIME
+            or execution_time > Options._MAX_EXECUTION_TIME
+        ):
+            raise ValueError(
+                f"max_execution_time must be between "
+                f"{Options._MIN_EXECUTION_TIME} and {Options._MAX_EXECUTION_TIME} seconds."
+            )
+
+    EnvironmentOptions.validate_environment_options(options.get("environment"))
+    ExecutionOptions.validate_execution_options(options.get("execution"))
+    SimulatorOptions.validate_simulator_options(options.get("simulator"))
+
+
+def _raise_if_error_in_options(options: Dict[str, Any]) -> None:
+    """Checks for settings that produce errors and raise a ValueError"""
+
+    # Fail on resilience_level set to 0
+    resilience_level = options.get("resilience_level", 1)
+    _check_argument(
+        resilience_level > 0,
+        description="Q-CTRL Primitives do not support resilience level 0. Please\
+        set resilience_level to 1 and re-try",
```

Comment:
```suggestion
        description=(
            "Q-CTRL Primitives do not support resilience level 0. Please "
            "set resilience_level to 1 and re-try"
        ),
```
minor nitpick but the error msg would print with a better format